pkg/utils/vmi.go

```diff
@@ -47,5 +52,36 @@ func (v *VmiGetter) WhoUseNad(nad *nadv1.NetworkAttachmentDefinition, nodesFilte
 		}
 	}
 
-	return afterFilter, nil
+	vmiNameList := make([]string, 0)
+	if len(afterFilter) > 0 {
+		for _, vmi := range afterFilter {
+			vmiNameList = append(vmiNameList, vmi.Namespace+"/"+vmi.Name)
+		}
+	}
+
+	vms, err := v.VMCache.GetByIndex(indexeres.VMByNetworkIndex, networkName)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, vm := range vms {
+		if vm.Namespace != nad.Namespace {
+			continue
+		}
+		vmiNameList = append(vmiNameList, vm.Namespace+"/"+vm.Name)
+	}
+
+	vmsTmp, err := v.VMCache.GetByIndex(indexeres.VMByNetworkIndex, nad.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, vm := range vmsTmp {
+		if vm.Namespace != nad.Namespace {
+			continue
```

Comment:
I am not sure about this existing check and if its needed, we can still attach a nad from any namespace to a vm in any namespace.This operation is supported currently.And with this check we might miss vms that have attached a nad from another namespace and the nad could be deleted while the vm is still present.